### PR TITLE
feat: pluggable issue tracker backend (GitHub + Linear)

### PR DESCRIPTION
**Pull request by sjawhar**
## Summary

Adds an `IssueTracker` interface so Legion can use either Linear or GitHub Issues (via GitHub Projects V2) as its issue tracking backend.

- **`IssueTracker` interface** with `parseIssues()` and `resolveTeamId()` — Linear and GitHub implementations
- **`POST /state/collect`** daemon endpoint replaces the stdin pipe for state analysis
- **`legion collect-state`** CLI command for debugging
- **`--backend github|linear`** flag on `legion start` (env var `LEGION_ISSUE_BACKEND`)
- **GitHub parser** handles real `gh project item-list --format json` output: `{items: [...]}` envelope, `owner/repo` repositories, cross-org projects, linked PR extraction
- **Structured `IssueSource`** metadata on `ParsedIssue` (owner, repo, number, url) — no lossy slug parsing
- **Session ID generalization** — accepts any string team ID, not just UUIDs
- **Skill files** updated for backend switching (controller, worker workflows, new `github/SKILL.md`)

## Smoke tested against real data

- **GitHub**: METR project (1506 items, 22 repos, 3 orgs) — 404 issues parsed, 152 with linked PRs
- **Linear**: LEG team (50 issues) — statuses, labels, PR refs from attachments all correct
- Both backends verified through daemon `POST /state/collect` and `legion collect-state` CLI

## Design docs

- `docs/plans/2026-02-16-github-issue-backend-design.md`
- `docs/plans/2026-02-16-github-issue-backend-impl.md`

## Test coverage

288 tests passing (43 new). Covers:
- GitHub parser: real schema, envelope unwrap, multi-org, PR extraction, null guards, edge cases
- Linear parser: GraphQL `{nodes}` format for attachments and labels, null guards
- `/state/collect` endpoint: both backends, error handling
- Session ID: non-UUID team IDs, golden values
- Config: backend validation, empty string rejection

### CODE REVIEW

File: packages/daemon/src/state/backends/github.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+      let labels: string[] = [];
+      if (Array.isArray(typedItem.labels)) {
+        labels = typedItem.labels.filter(
+          (label): label is string => typeof label === "string" && label !== ""
+        );
+      }
+
+      let prRef: GitHubPRRefType | null = null;
+      const linkedPRs = typedItem["linked pull requests"];
```
Comment (sjawhar) [inline packages/daemon/src/state/backends/github.ts]: High: parser currently only checks the linked pull requests key in one exact shape. Consider defensive support for common key-shape variants to avoid silent misses.

File: packages/daemon/src/daemon/server.ts [hunk 2]
```diff
@@ -337,6 +345,36 @@ export function startServer(opts: ServerOptions): { server: Server; stop: () =>
...
+          } catch {
+            return badRequest("invalid_json");
+          }
+
+          const backend = payload.backend;
+          if (!isBackendName(backend)) {
+            return badRequest("invalid_backend");
+          }
+
+          const issues = payload.issues;
```
Comment (sjawhar) [inline packages/daemon/src/daemon/server.ts]: Non-blocking: issues validation only rejects null and undefined. Tightening payload shape checks would prevent malformed input from silently producing empty-state success.

File: packages/daemon/src/state/fetch.ts [hunk 3]
```diff
@@ -278,169 +276,86 @@ export async function getPrDraftStatusBatch(
...
+      } catch {
+        for (const issueId of Object.keys(prRefsForStatus)) {
+          prDraftMap[issueId] = null;
+        }
+      }
+    })(),
+  ]);
 
-  for (const issue of parsedIssues) {
+  return parsedIssues.map((issue) => {
```
[the remark this answers is not in the document]
Comment (sjawhar) [reply packages/daemon/src/state/fetch.ts]: Fixed in the same change — `enrichParsedIssues` now maps `source: issue.source` into the `FetchedIssueData` result.

File: packages/daemon/src/state/backends/issue-tracker.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * Implementations parse raw issue data from their respective APIs
+ * into the normalized ParsedIssue format used by the state machine.
+ */
+export interface IssueTracker {
+  /** Parse raw issue data from the tracker into normalized form. */
+  parseIssues(raw: unknown): ParsedIssue[];
+
+  /** Resolve a team/project reference to a stable internal ID. */
+  resolveTeamId(ref: string): Promise<string>;
```
[the remark this answers is not in the document]
Comment (sjawhar) [reply packages/daemon/src/state/backends/issue-tracker.ts]: Removed from interface. Neither backend implements it and nothing calls it — the standalone `team-resolver.ts` handles CLI resolution. Can re-add when there's an actual consumer.

File: packages/daemon/src/state/backends/issue-tracker.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import type { ParsedIssue } from "../types";
+
+/**
+ * Pluggable issue tracker backend interface.
+ *
+ * Implementations parse raw issue data from their respective APIs
+ * into the normalized ParsedIssue format used by the state machine.
+ */
+export interface IssueTracker {
```
[the remark this answers is not in the document]
Comment (sjawhar) [reply packages/daemon/src/state/backends/issue-tracker.ts]: Added JSDoc to the `IssueTracker` interface documenting expected raw input shapes: Linear (array of GraphQL nodes) and GitHub (array or `{ items: [...] }` envelope). Also documented that `parseIssues` silently skips malformed entries.

File: packages/daemon/src/state/backends/issue-tracker.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (sjawhar) [reply packages/daemon/src/state/backends/issue-tracker.ts]: Removed from interface (YAGNI). Neither backend implemented it, nothing calls it. The CLI `resolveTeamId` in team-resolver.ts is a separate function.

File: .opencode/skills/legion-controller/SKILL.md [hunk 6]
```diff
@@ -50,26 +51,35 @@ digraph controller {
 ### 1. Fetch Issues
 
 ```bash
-LINEAR_JSON=$(linear_linear(action="search", query={"team": "$LINEAR_TEAM_ID"}))
+# Fetch issues based on backend
+if [ "$LEGION_ISSUE_BACKEND" = "github" ]; then
+  ISSUES_JSON=$(gh project item-list $PROJECT_NUM --owner $OWNER --format json)
```
[the remark this answers is not in the document]
Comment (sjawhar) [reply .opencode/skills/legion-controller/SKILL.md]: Fixed. `OWNER` and `PROJECT_NUM` are now derived from `LEGION_TEAM_ID` (format: `owner/project-number`) using parameter expansion before the fetch block.

File: .opencode/skills/AGENTS.md [hunk 7]
```diff
@@ -41,17 +44,18 @@ Set by daemon when spawning workers, consumed by skills:
```
[the remark this answers is not in the document]
Comment (sjawhar) [reply .opencode/skills/AGENTS.md]: Docs rewritten. Env vars are process-level for the controller — workers receive all context (issue ID, mode, backend) via the dispatch prompt, not env vars. Removed `LEGION_ISSUE_ID` and `LEGION_ISSUE_BACKEND` from worker env table.

File: packages/daemon/src/daemon/server.ts
[the remark this answers is not in the document]
Comment (sjawhar) [reply packages/daemon/src/daemon/server.ts]: Removed `env` from `/workers` endpoint entirely. Workers share the serve process environment; per-worker context (issue ID, mode, backend) is passed in the dispatch prompt. Daemon AGENTS.md updated to match.
